modules/statics/src/account.ts

```diff
@@ -95,12 +95,36 @@ export class ContractAddressDefinedToken extends AccountCoinToken {
   }
 }
 
+/**
+ * ERC20 token addresses are Base58 formatted on some blockchains.
+ */
+export class Base58ContractAddressDefinedToken extends AccountCoinToken {
+  public contractAddress: ContractAddress;
+
+  constructor(options: Erc20ConstructorOptions) {
+    super({
+      ...options,
+    });
+
+    if (!options.contractAddress.match(/[1-9A-HJ-NP-Za-km-z]{34}$/)) {
```

Comment:
I think you're trying to check if the full string match with the regex argument.
If that's the case, I'd use `/^[1-9A-HJ-NP-Za-km-z]{7}$/` instead of `/[1-9A-HJ-NP-Za-km-z]{7}$/` (see ^ at the beginning means the start of the string)

Example:
`'1AJNPa'.match(/[1-9A-HJ-NP-Za-km-z]{7}$/) // Returns true`
`'prefix1AJNPasufix'.match(/[1-9A-HJ-NP-Za-km-z]{7}$/) // Returns true too`
while
`'1AJNPa'.match(/^[1-9A-HJ-NP-Za-km-z]{7}$/) // Returns true`
`'prefix1AJNPasufix'.match(/^[1-9A-HJ-NP-Za-km-z]{7}$/) // Returns false`

Also, I think `regex.test` is more performant than `string.match` (https://stackoverflow.com/questions/10940137/regex-test-v-s-string-match-to-know-if-a-string-matches-a-regular-expression)

Reply:
Thank you Santi! I've modified it.